sql/core/src/main/scala/org/apache/spark/sql/execution/GeneratedAggregate.scala

```diff
@@ -270,7 +270,9 @@ case class GeneratedAggregate(
 
       val joinedRow = new JoinedRow3
 
-      if (groupingExpressions.isEmpty) {
+      if (!iter.hasNext) {
+        Iterator[InternalRow]()
```

Comment:
Can we add a comment here to explain why we're doing this?  Just needs to mention the memory leak issue.